lite/verifying/provider.go

```diff
@@ -0,0 +1,440 @@
+package verifying
```

Comment:
Should we rename the package to verifier?

Reply:
`verifying` makes more sense to me